Feature/3 return tika metadata

s4zuk3 opened the pull request:
Here is the new revised version of the Tika Metadata implementation. Any comments and/or changes are welcome.

Once again, thanks to @nmammeri and @KapiWow for their help with the HashMap configuration. And thank you for the opportunity to contribute a small part to this great project!

KapiWow commented on extractous-core/src/extractor.rs:
```diff
@@ -208,9 +221,13 @@ mod tests {
         let extractor = Extractor::new();
-        let result = extractor.extract_file_to_string(TEST_FILE);
-        let content = result.unwrap();
+        let result = extractor.extract_file_to_string_with_metadata(TEST_FILE);
```
It is better to add a separate test

s4zuk3 replied:
Done!

KapiWow reviewed (commented):
Looks good! Could you please also add metadata for parsing file, bytes, url. All of them are almost the same and use StreamReader, we can add metadata there.

nmammeri commented:
Was reviewing this today and found out that Tika metadata is actually Map<String, List<String>>. For example the key "pdf:charsPerPage" returns the number of characters per page for all pages inside the document. The current implementation returns only the first element and ignores the rest. 

I'm halfway through making the necessary changes by adding apache.tika.Meatada to StringResult and ReaderResult then converting to HashMap<String, Vec<String>> from Rust. the java parseMeatada() function is redandunt because we can loop through the metada from the rust side

s4zuk3 commented on bindings/extractous-python/src/extractor.rs:
```diff
@@ -196,3 +211,11 @@ impl Extractor {
+fn hashmap_to_pydict<'py>(py: Python<'py>, hashmap: &HashMap<String, String>) -> &'py PyDict {
+    let pydict = PyDict::new(py);
+    for (key, value) in hashmap {
+        pydict.set_item(key, value).unwrap();
```
Done!

s4zuk3 commented:
Hey! Thanks for the recommendations, I applied the changes that were instructed. I changed the Java step from HashMap to Tika.Metadata, fixed the HashMap to <String, Vec<String>>, and extended the functionality for all the other extraction methods.

Please let me know if any additional changes are needed, Thanks again @nmammeri  and @KapiWow .

nmammeri commented:
Thanks again @s4zuk3 for your the updates. amazing work. I've just made the checking logic of metadata tests more stringent. made it so 

In previous code if I remove some values of lists it doesn't detect it. Also If I remove some keys from the extracted metadata it does not get detected.

I changed it so that we check that all keys in expected metadata are found and the values for every expected key are similar to the extracted values.
